src/redux/thunks/previewCluster.ts

```diff
@@ -95,3 +113,108 @@ export const repreviewCluster = createAsyncThunk<
     state: RootState;
   }
 >('main/repreviewCluster', previewClusterHandler);
+
+/**
+ * Find the default version in line with the algorithm described at
+ * https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#version-priority
+ */
+
+const crdVersionRegex = /(v)(\d*)(alpha|beta)?(\d*)?/;
+
+function findDefaultVersion(r: K8sResource) {
+  const versionNames: string[] = r.content.spec.versions.map((v: any) => v.name);
+
+  versionNames.sort((a, b) => {
+    const m1 = crdVersionRegex.exec(a);
+    const m2 = crdVersionRegex.exec(b);
+
+    // do both versions match the regex?
+    if (m1 && m2) {
+      // do both have initial version number?
+      if (m1[2] && m2[2]) {
+        // is the initial version the same?
+        if (m1[2] === m2[2]) {
+          // do both have an alpha or beta tag?
+          if (m1[3] && m2[3]) {
+            // is the tag the same?
+            if (m1[3] === m2[3]) {
+              // do both have an alpha or beta version?
+              if (m1[4] && m2[4]) {
+                return parseInt(m1[4], 10) - parseInt(m2[4], 10);
+              }
+              return m1[4] ? 1 : -1;
+            }
+            // compare tags (negate for beta > alpha)
+            return -m1[3].localeCompare(m2[3]);
+          }
+          return m1[3] ? 1 : -1;
+        }
+        // compare version numbers
+        return parseInt(m1[2], 10) - parseInt(m2[2], 10);
+      }
+      return m1[2] ? 1 : -1;
+    }
+    if (m1) return 1;
+    if (m2) return -1;
+
+    return a.localeCompare(b);
+  });
+
+  return versionNames.length > 0 ? versionNames[0] : undefined;
+}
+
+/**
+ * Load custom resource objects for CRDs found in cluster
+ */
+
+async function loadCustomResourceObjects(
+  kc: KubeConfig,
+  customResourceDefinitions: K8sResource[],
+  configPath: string,
+  previewResult: SetPreviewDataPayload,
+  resourceRefsProcessingOptions: ResourceRefsProcessingOptions
+) {
+  const k8sApi = kc.makeApiClient(k8s.CustomObjectsApi);
+
+  try {
+    const customObjects = customResourceDefinitions.map(r =>
+      // retrieve objects using latest version name
+      // @ts-ignore
+      k8sApi
+        .listClusterCustomObject(r.content.spec.group, findDefaultVersion(r) || 'v1', r.content.spec.names.plural)
+        .then(response =>
+          // @ts-ignore
+          getK8sObjectsAsYaml(response.body.items)
+        )
+    );
+
+    if (!previewResult.previewResources) {
+      previewResult.previewResources = {};
+    }
+
+    await Promise.allSettled(customObjects).then(
```

Comment:
No need to use `then` if it is async/await

```
const customResults = await Promise.allSettled(customObjects);
(inside of then)
```

Reply:
thanks @erdkse  - not sure what you mean - allSettled returns a promise  - it would still have to be 

```
const customResultsPromise = Promise.allSettled(customObjects);
customResultsPromise.then(
  customResults => { .. },
  reason => { ... }
);
```
!?

Reply:
I'll try that - thanks!

Reply:
Yes, the alternative would be:
```try {
  const customResults = await Promise.allSettled(customObjects);
  // do something with customResults...
} catch(reason: any) {
}```